### PR TITLE
adding logs to log broadcaster

### PR DESCRIPTION
**Pull request by jinhoonbang**

### CODE REVIEW

**Comment by makramkd** [comment]
Should we make log level `Debug` rather than `Info`?

**Comment by jinhoonbang** [comment]
> Should we make log level `Debug` rather than `Info`?

yep that makes sense. Updated!

File: core/chains/evm/log/registrations.go [hunk 1]
```diff
@@ -410,6 +410,10 @@ func (r *handler) sendLog(log types.Log, latestHead evmtypes.Head,
 		if len(filters) > 0 && len(log.Topics) > 1 {
 			topicValues := log.Topics[1:]
 			if !filtersContainValues(topicValues, filters) {
+				logger.Debugw("filters did not contain expected topic",
+					"blockNumber", log.BlockNumber, "blockHash", log.BlockHash,
+					"address", log.Address, "latestBlockNumber", latestBlockNumber,
+					"topicValues", topicValues, "filters", filters)
```
Comment (makramkd) [inline core/chains/evm/log/registrations.go:416]: What is the type of `filters` here? Is it `[]common.Hash`?

Comment (makramkd) [reply core/chains/evm/log/registrations.go:416]: If it's `[][]byte` then this log won't be very digestable, we'd need to hex-ify it.

Comment (jinhoonbang) [reply core/chains/evm/log/registrations.go:416]: yeah filters is `[][]Topic`, where Topic is `common.Hash`. Updated the code to hexify it.

**Review by makramkd — APPROVED** [review]

File: core/chains/evm/log/broadcaster.go [hunk 2]
```diff
@@ -510,9 +510,11 @@ func (b *broadcaster) onNewLog(log types.Log) {
 
 	if log.Removed {
 		// Remove the whole block that contained this log.
+		b.logger.Debugw("found reverted log", "log", log)
```
Comment (samsondav) [inline core/chains/evm/log/broadcaster.go:513]: Please capitalise log messages.

File: core/chains/evm/log/broadcaster.go [hunk 3]
```diff
@@ -510,9 +510,11 @@ func (b *broadcaster) onNewLog(log types.Log) {
...
 		b.logPool.removeBlock(log.BlockHash, log.BlockNumber)
 		return
 	} else if !b.registrations.isAddressRegistered(log.Address) {
+		b.logger.Debugw("found unregistered address", "Address", log.Address)
```
Comment (samsondav) [inline core/chains/evm/log/broadcaster.go:517]: keys should not be capitalised

File: core/chains/evm/log/pool.go [hunk 4]
```diff
@@ -50,14 +52,16 @@ type logPool struct {
 	// This min-heap maintains block numbers of logs in the pool.
 	// it helps us easily determine the minimum log block number
 	// in the pool (while the set of log block numbers is dynamically changing).
-	heap *pairingHeap.PairHeap
+	heap   *pairingHeap.PairHeap
+	logger logger.Logger
 }
 
-func newLogPool() *logPool {
+func newLogPool(logger logger.Logger) *logPool {
```
Comment (samsondav) [inline core/chains/evm/log/pool.go:59]: Please call the variable name `lggr`, for consistency with other places in the code.

**Review by samsondav — CHANGES_REQUESTED** [review]

**Review by makramkd — APPROVED** [review]

**Review by samsondav — APPROVED** [review]

**Pull request merged**